File: Benchmark/benchmark.cpp
#include <QActiveResource.h>

int main(int argc, char *argv[])
{
    Q_ASSERT(getenv("AR_BASE") && getenv("AR_RESOURCE") && getenv("AR_FIELD"));

    int count = 100;

    if(argc > 1)
    {
        count = QString(argv[1]).toInt();
    }

    QActiveResource::Resource resource(QUrl(getenv("AR_BASE")), getenv("AR_RESOURCE"));

    const QString field = getenv("AR_FIELD");

    for(int i = 1; i <= count; i++)
    {
        printf("%i\n", i);

        foreach(QActiveResource::Record record, resource.find())
        {
            printf("%s\n", record[field].toString().toUtf8().data());
        }
    }

    return 0;
}

File: Benchmark/benchmark.rb
#!/usr/bin/ruby

require 'rubygems'
require 'active_resource'

type = ARGV.shift || 'REXML'
count = ARGV.shift || 100

case type
when 'nokogiri'
  require 'nokogiri'
  ActiveSupport::XmlMini.backend = 'Nokogiri'
when 'qar'
  require 'QAR'
  extend = 'extend QAR'
end

resource = ENV['AR_RESOURCE'].capitalize

eval "class #{resource} < ActiveResource::Base ; #{extend} ; self.site = ENV['AR_BASE'] ; end"

resource = Kernel.const_get(resource)
field = ENV['AR_FIELD']

unless type == 'dummy'
  (1..count).each { |i| puts i ; resource.find(:all).each { |p| puts p.send(field) } }
end
